### PR TITLE
BAU: Use the new metadata published by the apps

### PR DESCRIPTION
**Pull request by CharlesIC**
- Changed entity IDs to match the new metadata endpoints
  - Rename secrets in K8s cluster to be more clear by better describing what purpose they serve
  - Add a separate metadata signing cert for Stub Connector

### CODE REVIEW

File: chart/templates/metadata-ingress-virtual-service.yaml [hunk 1]
```diff
@@ -16,6 +16,6 @@ spec:
   http:
     - route:
       - destination:
-          host: {{ .Release.Name }}-proxy-metadata
+          host: {{ .Release.Name }}-proxy-node-metadata
```
Comment (blairboy362) [inline chart/templates/metadata-ingress-virtual-service.yaml]: This is pointing at the service that is created by the VMC when processing a `Metadata` resource, which might not be what we want. Potentially lengthy discussion to have around this that I'm not going to attempt via github comments.

Comment (CharlesIC) [reply chart/templates/metadata-ingress-virtual-service.yaml]: Actually we don't need this anymore - that's the thing I was talking about this morning. Let's discuss tomorrow.

File: chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 {{- define "connector.host" -}}
```
Comment (blairboy362) [inline chart/templates/_helpers.tpl]: Thinking about it...this block can be simplified further. We no longer need the `if...else`; just the `printf` inside the `if` part.

File: chart/templates/_helpers.tpl [hunk 3]
```diff
@@ -5,7 +5,7 @@
...
 {{- if .Values.stubConnector.enabled -}}
```
Comment (blairboy362) [inline chart/templates/_helpers.tpl]:
```suggestion
```

File: chart/templates/_helpers.tpl [hunk 4]
```diff
@@ -5,7 +5,7 @@
...
-{{- printf "%s-%s.%s" .Release.Name "connector" (required "global.cluster.domain is required" .Values.global.cluster.domain) | trimSuffix "-" -}}
+{{- printf "%s" (include "connector.metadata.host" .) | trimSuffix "-" -}}
 {{- else -}}
```
Comment (blairboy362) [inline chart/templates/_helpers.tpl]:
```suggestion
```

File: chart/templates/_helpers.tpl [hunk 5]
```diff
@@ -5,7 +5,7 @@
...
 {{- printf "%s" .Values.connector.host -}}
```
Comment (blairboy362) [inline chart/templates/_helpers.tpl]:
```suggestion
```

File: chart/templates/_helpers.tpl [hunk 6]
```diff
@@ -5,7 +5,7 @@
...
 {{- end -}}
```
Comment (blairboy362) [inline chart/templates/_helpers.tpl]:
```suggestion
```

**Comment by blairboy362** [comment]
You may also be able to remove "metadata.host" from the `_helpers.tpl` (confirm it with a search, but I hope it is unused now).

**Review by blairboy362 — APPROVED** [review]

**Review by idrop — APPROVED** [review]

**Pull request merged**